jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt

```diff
@@ -23,7 +23,7 @@ class CodeWhispererEditorListener : EditorFactoryListener {
                     CodeWhispererInvocationStatus.getInstance().documentChanged()
                     editor.project?.let { project ->
                         PsiDocumentManager.getInstance(project).getPsiFile(editor.document)?.codeWhispererLanguage ?. let { language ->
-                            CodeWhispererCodeCoverageTracker.getInstance(language).apply {
+                            CodeWhispererCodeCoverageTracker.getInstance(language)?.apply {
                                 activateTrackerIfNotActive()
                                 documentChanged(event)
```

Comment:
Since we are intoducing support to only certain languages here we might as well apply this condition in this editorCreated() method. If the language is not supported, we don't even need to listen for documentChanged() event for that document.

Reply:
In this way we also don't need to make getInstance() to be a nullable

Reply:
sounds good let me update this